types/TestResult.js

```diff
@@ -100,20 +100,20 @@ export type FormattedAssertionResult = {
 
 export type AggregatedResult = {
   coverageMap?: ?CoverageMap,
-  numFailedTests: number,
-  numFailedTestSuites: number,
-  numPassedTests: number,
-  numPassedTestSuites: number,
-  numPendingTests: number,
-  numPendingTestSuites: number,
-  numRuntimeErrorTestSuites: number,
-  numTotalTests: number,
-  numTotalTestSuites: number,
-  snapshot: SnapshotSummary,
-  startTime: number,
+  numFailedTests?: number,
+  numFailedTestSuites?: number,
+  numPassedTests?: number,
+  numPassedTestSuites?: number,
+  numPendingTests?: number,
+  numPendingTestSuites?: number,
+  numRuntimeErrorTestSuites?: number,
+  numTotalTests?: number,
+  numTotalTestSuites?: number,
+  snapshot?: SnapshotSummary,
+  startTime?: number,
   success: boolean,
-  testResults: Array<TestResult>,
-  wasInterrupted: boolean,
+  testResults?: Array<TestResult>,
+  wasInterrupted?: boolean,
```

Comment:
Yeah, I think that makes sense.

If all else fails, you could type onComplete as `AggregatedResult | {success: boolean}`. It probably makes more sense to fill it out though.